modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/distruptor/inbound/QueuePurgeEvent.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.andes.kernel.distruptor.inbound;
+
+import com.google.common.util.concurrent.SettableFuture;
+
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Class to hold information relevant to queue purging event
+ */
+public class QueuePurgeEvent {
+
+    /**
+     * queue or topic name (subscribed routing key) whose messages should be removed
+     */
+    private String destination;
+
+    /**
+     * The user who initiated the purge request
+     */
+    private String ownerName;
+
+    /**
+     * weather purging happens for a topic
+     */
+    private boolean isTopic;
+
+    /**
+     * Purged message count as a future
+     */
+    SettableFuture<Integer> purgedCount;
```

Comment:
Explain the thought process behind choosing SettableFuture here. (that will add value to the documentation)